sdk/devcenter/developer-devcenter-rest/test/public/devBoxTests.spec.ts

```diff
@@ -0,0 +1,541 @@
+      const { name } = action;
+      console.log(`Received DevBox action"${name}"`);
+      actions.push(action);
+    }
+
+    expect(actions.length).to.equal(1);
+    expect(actions[0].name).to.equal("schedule-default");
+
+    await deleteDevBox();
+  });
+
+  it("GetAction", async function () {
+    await createDevBox();
+
+    const actionResponse = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}/actions/{actionName}",
+        projectName,
+        userId,
+        devboxName,
+        "schedule-default"
+      )
+      .get();
+
+    if (isUnexpected(actionResponse)) {
+      throw actionResponse.body.error;
+    }
+
+    expect(actionResponse.body.name).to.equal("schedule-default");
+
+    await deleteDevBox();
+  });
+
+  it("SkipAction", async function () {
+    await createDevBox();
+
+    const skipActionResponse = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}/actions/{actionName}:skip",
+        projectName,
+        userId,
+        devboxName,
+        "schedule-default"
+      )
+      .post();
+
+    assert.equal(skipActionResponse.status, "204");
+
+    await deleteDevBox();
+  });
+
+  it("DelayAction", async function () {
+    await createDevBox();
+
+    const delayActionParameters: DevBoxesDelayAllActionsParameters = {
+      queryParameters: {
+        until: "2023-05-06T00:00:00Z",
+      },
+    };
+    
+    const delayActionResponse = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}/actions/{actionName}:delay",
+        projectName,
+        userId,
+        devboxName,
+        "schedule-default"
+      )
+      .post(delayActionParameters);
+
+    assert.equal(
+      delayActionResponse.status,
+      "200",
+      "Delaying DevBox action should return 200 OK."
+    );
+    
+    await deleteDevBox();
+  });
+
+  it("DelayAllActions", async function () {
+    await createDevBox();
+
+    const delayActionsParameters: DevBoxesDelayAllActionsParameters = {
+      queryParameters: {
+        until: "2023-05-06T00:00:00Z",
+      },
+    };
+
+    const delayActionsResponse = await client
+      .path(
+        "/projects/{projectName}/users/{userId}/devboxes/{devBoxName}/actions:delay",
+        projectName,
+        userId,
+        devboxName
+      )
+      .post(delayActionsParameters);
+
+    if (isUnexpected(delayActionsResponse)) {
+      throw new Error(delayActionsResponse.body?.error.message);
+    }
+
+    const actionDelayResults: DevBoxActionDelayResultOutput[] = [];
+
+    for await (const actionDelay of paginate(client, delayActionsResponse)) {
+      const { name } = actionDelay;
+      actionDelayResults.push(actionDelay);
+    }
+
+    expect(actionDelayResults.length).to.equal(1);
+    expect(actionDelayResults[0].result).to.equal("Succeeded");
+
+    await deleteDevBox();
+  });
+
+  it("StartDevBox", async function () {
+    await createDevBox();
```

Comment:
Same as in the .NET pull request, I think it would be more meaningful to also stop this dev box after creation. Start will be a no-op if it's already in a running state.

Reply:
Fixed here: https://github.com/Azure/azure-sdk-for-js/pull/25859/commits/a2ee43b25b8a192d11e4d708f46f5b40cdbcb2a8